File: c/LeetCode_11_ContainerWithMostWater.c
#include <stdio.h>
#include <stdlib.h>


int maxArea(int* height, int heightSize) {
    int low = 0;
    int high = heightSize - 1;
    int max_volume = 0;
    while (low < high) {
        int m_min = (height[low] > height[high]) ? height[high] : height[low];
        int volume = (high - low) * m_min;
        if (volume > max_volume) {
            max_volume = volume;
        }

        if (height[low] > height[high]) {
            --high;
        } else {
            ++low;
        }
    }

    return max_volume;
}


int main() {

    return 0;
}

File: python/LeetCode_11_ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        low = 0
        high = len(height) - 1
        max_volume = 0
        while (low < high):
            min_m = min(height[low], height[high])
            volume = (high - low) * min_m
            if (volume > max_volume):
                max_volume = volume

            if (height[low] > height[high]):
                high -= 1
            else:
                low += 1
        return max_volume